sklearn/ensemble/tests/test_forest.py

```diff
@@ -1307,6 +1310,70 @@ def test_forest_degenerate_feature_importances():
                        np.zeros(10, dtype=np.float64))
 
 
+def test_forest_balanced_bootstrap_not_implemented():
+    # check that an error is raised for unknown target
+    X, y = make_multilabel_classification(random_state=0)
+    with pytest.raises(NotImplementedError, match="Balanced random-forest"):
+        RandomForestClassifier(class_weight="balanced_bootstrap").fit(X, y)
+
+
+def test_forest_class_weight_balanced_bootstrap():
+    # check the implementation of balanced random forest
+    X, y = datasets.make_classification(
+        n_samples=1000, weights=[0.95, 0.05], random_state=0
+    )
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, stratify=y, random_state=0
+    )
+
+    rf = RandomForestClassifier(n_estimators=10, random_state=0)
+    brf = RandomForestClassifier(
+        n_estimators=10, random_state=0, class_weight="balanced_bootstrap"
+    )
+
+    rf.fit(X_train, y_train)
+    brf.fit(X_train, y_train)
+
+    # The random-forest will be affected by the imbalanced classes while the
+    # balanced random-forest will be able to fit a proper model
+    balanced_acc_rf = balanced_accuracy_score(y_test, rf.predict(X_test))
+    balanced_acc_brf = balanced_accuracy_score(y_test, brf.predict(X_test))
+    assert balanced_acc_brf > balanced_acc_rf
+
+    # the balanced random-forest will be trained with much less samples than
+    # the random-forest. Thus, the number of nodes will be much less.
+    for brf_est, rf_est in zip(brf.estimators_, rf.estimators_):
+        assert brf_est.tree_.node_count < rf_est.tree_.node_count
+
+
+def test_forest_balanced_bootstrap_max_samples():
+    # check that we take the minimum between max_samples and the minimum
```

Comment:
we could also have a test where `max_samples` is set in two models in a way that the outcome is identical in both cases.